apps/emqx/test/emqx_persistent_session_SUITE.erl

```diff
@@ -774,17 +774,20 @@ t_publish_many_while_client_is_gone_qos1(Config) ->
 
     ?assert(NMsgs2 < NPubs, {NMsgs2, '<', NPubs}),
     ?assert(NMsgs2 > NPubs2, {NMsgs2, '>', NPubs2}),
-    ?assert(NMsgs2 >= NPubs - NAcked, Msgs2),
-    NSame = NMsgs2 - NPubs2,
+    %% Once reconnected, we should receive all the published messages above (`Msgs1 ++
+    %% Msgs2'), *except* those from `Msgs1' that we acked.
```

Comment:
That is: we must not get the same messages which we had already acked, but we may get fewer than all the published and not-yet-acked messages.